src/main/java/seedu/address/logic/commands/SetCommand.java

```diff
@@ -53,15 +53,32 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Ingredient> lastShownList = model.getFilteredIngredientList();
 
-        Ingredient ingredientToEdit = lastShownList.get(lastShownList.indexOf(target));
-        Ingredient updatedIngredient = createSetIngredient(target, setIngredientDescriptor);
+        Ingredient ingredientToEdit = null;
+        boolean isNoChange = false;
 
-        if (!ingredientToEdit.isSameIngredient(updatedIngredient)
-                && model.hasIngredient(updatedIngredient)) {
+        for (Ingredient ingredient : lastShownList) {
+            if (ingredient.getIngredientName().equals(targetName)) {
+                ingredientToEdit = ingredient;
+                if (ingredientToEdit.getAmount().equals(setIngredientDescriptor.getAmount().get())) {
```

Comment:
You may want to change this part to if (AAA && BBB) to avoid the deep nesting? Thanks

Reply:
good point, changing now

Reply:
noted, editing now

Reply:
👍